pyocd/debug/svd/parser.py

```diff
@@ -0,0 +1,448 @@
+#
+# Copyright 2015 Paul Osborne <osbpau@gmail.com>
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#     http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+#
+from xml.etree import ElementTree as ET
+
+import six
+
+from .model import SVDDevice
+from .model import SVDPeripheral
+from .model import SVDInterrupt
+from .model import SVDAddressBlock
+from .model import SVDRegister, SVDRegisterArray
+from .model import SVDRegisterCluster, SVDRegisterClusterArray
+from .model import SVDField
+from .model import SVDEnumeratedValue
+from .model import SVDCpu
+import pkg_resources
+import re
+
+
+def _get_text(node, tag, default=None):
+    """Get the text for the provided tag from the provided node"""
+    try:
+        return node.find(tag).text
+    except AttributeError:
+        return default
+
+
+def _get_int(node, tag, default=None):
+    text_value = _get_text(node, tag, default)
+    try:
+        if text_value != default:
+            text_value = text_value.strip().lower()
+            if text_value.startswith('0x'):
+                return int(text_value[2:], 16)  # hexadecimal
+            elif text_value.startswith('#'):
+                # TODO(posborne): Deal with strange #1xx case better
+                #
+                # Freescale will sometimes provide values that look like this:
+                #   #1xx
+                # In this case, there are a number of values which all mean the
+                # same thing as the field is a "don't care".  For now, we just
+                # replace those bits with zeros.
+                text_value = text_value.replace('x', '0')[1:]
+                is_bin = all(x in '01' for x in text_value)
+                return int(text_value, 2) if is_bin else int(text_value)  # binary
+            elif text_value.startswith('true'):
+                return 1
+            elif text_value.startswith('false'):
+                return 0
+            else:
+                return int(text_value)  # decimal
+    except ValueError:
+        return default
+    return default
+
+
+class SVDParser(object):
+    """The SVDParser is responsible for mapping the SVD XML to Python Objects"""
+
+    @classmethod
+    def for_xml_file(cls, path, remove_reserved=False):
+        return cls(ET.parse(path), remove_reserved)
+
+    @classmethod
+    def for_packaged_svd(cls, vendor, filename, remove_reserved=False):
```

Comment:
If we're going to drop non-Pack-based SVD files, you can remove `for_packaged_svd()`.

We could consider using `cmsis_svd` only as a data source for builtin targets and keeping this method. That would require bringing back `is_local` in `SVDFile`.